engine/src/main/java/com/datatorrent/stram/api/StramEvent.java

```diff
@@ -67,6 +68,23 @@ public void setReason(String reason)
     this.reason = reason;
   }
 
+  public void setLogLevel(LogLevel logLevel)
```

Comment:
I don't see a need for the setter method. Even if event type will not be associated with the log level in the future, the log level should be known when an event is created. Provide default constructor (the one that does not take log level and uses default event type log level) and optionally another constructor that takes log level.